src/main/java/frc/robot/subsystems/DriveTrain.java

```diff
@@ -309,59 +302,59 @@ public void periodic() {
 
         SmartDashboard.putBoolean("drivetrain/fieldCentric", m_fieldCentric);
 
-		for (SwerveModule mod : m_swerveModules) {
-			mod.updateSmartDashboard();
-		}
-	}
+        for (SwerveModule mod : m_swerveModules) {
+            mod.updateSmartDashboard();
+        }
+    }
 
     // get the trajectory following autonomous command in PathPlanner using the name
     public Command getTrajectoryFollowingCommand(String trajectoryName) {
 
         PathPlannerTrajectory traj = PathPlanner.loadPath(trajectoryName, 2.0, 1.0);
 
-        Command command = new FollowTrajectory(
-                this,
-                traj,
-                () -> this.getPose(),
-                m_kinematics,
-                m_xController,
-                m_yController,
-                m_thetaController,
-                (states) -> {
+        Command command = new FollowTrajectory(this, traj, () -> this.getPose(), m_kinematics, m_xController,
+                m_yController, m_thetaController, (states) -> {
                     this.drive(m_kinematics.toChassisSpeeds(states));
-                },
-                this).andThen(() -> stop());
+                }, this).andThen(() -> stop());
 
         return command;
     }
 
-    //get target pose
-    public Pose2d getTargetPose(){
-        return m_poseTest;
+    // get best pose to shoot based on Apriltag pose, the pose you want to go to
+    public Pose2d getTagRobotPose(boolean shiftLeft, boolean shiftRight) {
+        if(m_vision.getCentralTagPose().isPresent()){
+            Pose2d tagPose = m_vision.getCentralTagPose().get(); // get AprilTag pose of target ID tag
```

Comment:
You are not handling the error. If the return value is null, this will fail. You need to test the value first.